tools/integration_tests/read_cache/helpers_test.go

```diff
@@ -84,21 +89,68 @@ func getCachedFilePath() string {
 	return path.Join(cacheLocationPath, cacheSubDirectoryName, setup.TestBucket(), testDirName, testFileName)
 }
 
-func (s *testStruct) validateFileInCacheDirectory(t *testing.T) {
+func validateFileSizeInCacheDirectory(filesize int64, t *testing.T) (expectedPathOfCachedFile string) {
 	// Validate that the file is present in cache location.
-	expectedPathOfCachedFile := getCachedFilePath()
+	expectedPathOfCachedFile = getCachedFilePath()
 	fileInfo, err := operations.StatFile(expectedPathOfCachedFile)
 	if err != nil {
 		t.Errorf("Failed to find cached file %s: %v", expectedPathOfCachedFile, err)
 	}
 	// Validate file size in cache directory matches actual file size.
-	if (*fileInfo).Size() != fileSize {
-		t.Errorf("Incorrect cached file size. Expected %d, Got: %d", fileSize, (*fileInfo).Size())
+	if (*fileInfo).Size() != filesize {
+		t.Errorf("Incorrect cached file size. Expected %d, Got: %d", filesize, (*fileInfo).Size())
 	}
+	return
+}
+
+func validateFileInCacheDirectory(filesize int64, ctx context.Context, storageClient *storage.Client, t *testing.T) {
+	expectedPathOfCachedFile := validateFileSizeInCacheDirectory(filesize, t)
 	// Validate content of file in cache directory matches GCS.
```

Comment:
Shouldn't we do this?
```
validateFileSizeInCacheDirectory(filesize, t)
expectedPathOfCachedFile := getCachedFilePath()
```

According to me returning filePath in size validation function does not seems correct?